packages/atomic/cypress/integration/result-list/result-list-utils.ts

```diff
@@ -0,0 +1,37 @@
+import {buildTestUrl, injectComponent} from '../../utils/setupComponent';
+import {
+  resultListComponent,
+  ResultListSelectors,
+} from './result-list-v1-selectors';
+
+export function withAnySectionnableResultList(
+  assertions: (setUpResultListPage: (template: string) => void) => void
+) {
+  const viewports = {mobile: 1023, desktop: 1024};
+  Object.entries(viewports).forEach(([viewport, width]) =>
+    describe(`with a ${viewport} viewport`, () =>
+      ['list', 'grid'].forEach((display) =>
+        describe(`in a result ${display}`, () =>
+          ['none', 'icon', 'small', 'large'].forEach((image) =>
+            describe(`with image="${image}"`, () =>
+              ['compact', 'normal', 'comfortable'].forEach((density) =>
+                describe(`with density="${density}"`, () => {
+                  function setUpResultListPage(template: string) {
+                    const aspectRatio = 16 / 9;
+                    cy.viewport(width, width / aspectRatio);
+                    cy.visit(buildTestUrl());
+                    cy.injectAxe();
+                    injectComponent(
+                      resultListComponent(template, display, image, density),
+                      true
+                    );
+                    cy.get('.list-wrapper:not(.placeholder)');
+                  }
+
+                  assertions(setUpResultListPage);
+                })
+              ))
+          ))
+      ))
```

Comment:
If lines 21-23 are done once (creating the viewport, visiting the url, injecting axe), and then in each test, only the result template is injected, assertions are run, followed by ejecting in preparation for the next test, could that bring the time down significantly?

I fully agree in having these tests to cover the different possibilities. However, if we can at the same time keep the execution time low, this will have a huge productivity benefit.

For example, a few months ago, cypress tests were taking ~5min, whereas today they are over ~10 min. This worries me because Atomic is still a new project, and I think about what the number will be in 6 months from now. If instead we proactively make optimizations as we go, it will help the project over the long term.

Reply:
I'll test that out

Reply:
I reduced repetition and now these tests take about 21 seconds to run in total instead of 76 seconds.

Reply:
That's great! 💯